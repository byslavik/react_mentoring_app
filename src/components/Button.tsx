import * as React from 'react';

import * as buttons from '../scss/buttons.scss';


interface ButtonState {
}
interface ButtonProps {
  action: any;
  isActive?: boolean;
  style?: string;
}

export class Button extends React.Component<ButtonProps, ButtonState> {
    render() {
      const style = this.props.style ? this.props.style : (buttons.buttonWhite + ' ' + buttons.small);

      return (<button className={style + (this.props.isActive ? ' ' + buttons.buttonWhiteActive : '')} onClick={this.props.action} >{this.props.children}</button>)
    }
}
