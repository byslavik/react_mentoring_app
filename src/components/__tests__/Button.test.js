import * as React from 'react';
import {Button} from '../Button.tsx';
import * as renderer from 'react-test-renderer';

describe('Button component', ()=>{
  it('Button component shows correctly', () => {
    const tree = renderer.create(
      <Button action={()=>{}} isActive={false} style="red">Button</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button gets active class', () => {
    const tree = renderer.create(
      <Button action={()=>{}} isActive={true} style="red">Button</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});