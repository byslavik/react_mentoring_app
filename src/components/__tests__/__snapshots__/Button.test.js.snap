// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button component shows correctly 1`] = `
<button
  className="red"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button component Button gets active class 1`] = `
<button
  className="red undefined"
  onClick={[Function]}
>
  Button
</button>
`;
